flow/ThreadHelper.actor.h

```diff
@@ -727,5 +727,76 @@ class ThreadSafeAsyncVar : NonCopyable, public ThreadSafeReferenceCounted<Thread
 	ThreadSpinLock lock;
 };
 
+// Like a future (very similar to ThreadFuture) but only for computations that already completed. Reuses the SAV's
+// implementation for memory management  error handling though. Essentially a future that's returned from a synchronous
+// computation and guaranteed to be complete.
+
+template <class T>
+class ThreadResult {
+public:
+	T get() { return sav->get(); }
+
+	bool isValid() const { return sav != 0; }
+	bool isError() { return sav->isError(); }
+	Error& getError() {
+		if (!isError())
+			throw future_not_error();
+
+		return sav->error;
+	}
+
+	void cancel() { extractPtr()->cancel(); }
+
+	ThreadResult() : sav(0) {}
+	explicit ThreadResult(ThreadSingleAssignmentVar<T>* sav) : sav(sav) {
```

Comment:
If we are going to allow construction via a `sav`, can we assert that the `sav` is in a ready state?

Reply:
Good idea